app/scripts/controllers/app-metadata.ts

```diff
@@ -12,19 +29,84 @@ export type AppMetadataControllerState = {
 };
 
 /**
- * The options that NetworkController takes.
+ * Function to get default state of the {@link AppMetadataController}.
  */
-export type AppMetadataControllerOptions = {
-  currentMigrationVersion?: number;
-  currentAppVersion?: string;
-  state?: Partial<AppMetadataControllerState>;
-};
+export const getDefaultAppMetadataControllerState =
+  (): AppMetadataControllerState => ({
+    currentAppVersion: '',
+    previousAppVersion: '',
+    previousMigrationVersion: 0,
+    currentMigrationVersion: 0,
+  });
+
+/**
+ * Returns the state of the {@link AppMetadataController}.
+ */
+export type AppMetadataControllerGetStateAction = ControllerGetStateAction<
+  typeof controllerName,
+  AppMetadataControllerState
+>;
+
+/**
+ * Actions exposed by the {@link AppMetadataController}.
+ */
+export type AppMetadataControllerActions = AppMetadataControllerGetStateAction;
+
+/**
+ * Event emitted when the state of the {@link AppMetadataController} changes.
+ */
+export type AppMetadataControllerStateChangeEvent = ControllerStateChangeEvent<
+  typeof controllerName,
+  AppMetadataControllerState
+>;
+
+export type AppMetadataControllerEvents = AppMetadataControllerStateChangeEvent;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions = never;
```

Comment:
Technically we should not export `AllowedActions` or `AllowedEvents`: https://github.com/MetaMask/core/blob/main/docs/writing-controllers.md#define-but-do-not-export-a-type-union-for-external-action-types. I realize that the example controller is incorrect in this regard. I will update this.

``` suggestion
type AllowedActions = never;
```